site/zenodo_rdm/files.py

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# ZenodoRDM is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Zenodo files utilities."""
+
+import mimetypes
+from urllib.parse import urlsplit, urlunsplit
+
+import requests
+from flask import current_app, make_response, request
+from invenio_files_rest.helpers import sanitize_mimetype
+from invenio_files_rest.storage.pyfs import pyfs_storage_factory
+
+try:
+    from invenio_xrootd.storage import EOSFileStorage as BaseFileStorage
+    from requests_kerberos import DISABLED, HTTPKerberosAuth
+except ImportError:
+    # fake requests_kerberos
+    HTTPKerberosAuth = type("obj", (object,), {})
+    DISABLED = 3
+    # use base PyFSFileStorage instead
+    from invenio_files_rest.storage.pyfs import PyFSFileStorage as BaseFileStorage
+
+
+class EOSFilesOffload(BaseFileStorage):
+    """Offload file downloads to another server."""
+
+    def _get_eos_redirect_path(self):
+        """Get the real path of the file streamed from another server."""
+        host = current_app.config["ZENODO_EOS_OFFLOAD_HTTPHOST"]
+        redirect_base_path = current_app.config["ZENODO_EOS_OFFLOAD_REDIRECT_BASE_PATH"]
+        base_path = urlsplit(self.fileurl).path
+        eos_resp = requests.get(
+            f"{host}/{base_path}",
+            auth=HTTPKerberosAuth(DISABLED),
+            verify=False,
+            allow_redirects=False,
+        )
+        assert eos_resp.status_code == 307
```

Comment:
```suggestion
        if eos_resp.status_code != 307:
            current_app.logger.error("EOS redirect failed")
            return
```

Maybe we return and fallback to serving through the app? I know it's a bit risky since if it starts happening permanently we won't notice (that's why I also proposed the error logging).